check-build-status/index.js

```diff
@@ -0,0 +1,113 @@
+const AWS = require("aws-sdk");
+const fs = require("fs");
+const createCsvWriter = require("csv-writer").createObjectCsvWriter;
+
+AWS.config.update({ region: "ap-southeast-1" });
+
+const amplify = new AWS.Amplify();
+
+// Function to get all Amplify apps and write to a JSON file
+const getAllApps = async () => {
+  try {
+    const apps = [];
+    let nextToken;
+    do {
+      const params = { nextToken };
+      const response = await amplify.listApps(params).promise();
+      apps.push(...response.apps);
+      nextToken = response.nextToken;
+    } while (nextToken);
+
+    const appData = apps.map((app) => ({ appId: app.appId, name: app.name }));
+    fs.writeFileSync("amplify_apps.json", JSON.stringify(appData, null, 2));
+
+    return appData;
+  } catch (error) {
+    console.error("Error fetching Amplify apps:", error);
+  }
+};
+
+// Function to check the build status of specific branches and write to CSV files incrementally
+const checkBuildStatuses = async (apps) => {
+  // Commenting this out to allow for faster processing
+  /* 
+  const successCsvWriter = createCsvWriter({
+    path: 'success.csv',
+    header: [
+      { id: 'appId', title: 'appId' },
+      { id: 'appName', title: 'appName' },
+      { id: 'branch', title: 'branch' },
+      { id: 'buildStatus', title: 'buildStatus' }
+    ],
+    append: true
+  });
+  */
+
+  const failedCsvWriter = createCsvWriter({
+    path: "failed.csv",
+    header: [
+      { id: "appId", title: "appId" },
+      { id: "appName", title: "appName" },
+      { id: "branch", title: "branch" },
+      { id: "buildStatus", title: "buildStatus" },
+    ],
+    append: true,
+  });
+
+  for (const app of apps) {
+    const branches = ["master", "staging", "staging-lite"];
+    for (const branchName of branches) {
+      try {
+        const response = await amplify
+          .listJobs({ appId: app.appId, branchName })
+          .promise();
```

Comment:
Nit: we can also use `maxResults: 1` since we only care about the latest build

Reply:
good point, added